packages/wallet/src/wallet.ts

```diff
@@ -327,4 +328,53 @@ export class Wallet {
         await this.db.saveSilentPaymentAddress(address);
         return address;
     }
+
+    async scanSilentBlock(silentBlock: SilentBlock): Promise<void> {
+        const dustLimit = 546;
+        const matchedUTXOs: Coin[] = [];
+    
+        const coinType = this.network.network.bech32 === 'bc' ? 0 : 1;
+        const spendKey = this.masterKey.derivePath(`m/352'/${coinType}'/0'/0'/0`);
+        
+        const spendPublicKey = spendKey.publicKey;
+    
+        for (const transaction of silentBlock.transactions) {
+            const outputs = transaction.outputs.filter((output) => output.value >= dustLimit);
+    
+            if (outputs.length === 0) continue;
+    
+            const outputPubKeys = outputs.map((output) => Buffer.from(output.pubKey, 'hex'));
+    
+            const scanTweak = Buffer.from(transaction.scanTweak, 'hex');
+    
+            const matches = scanOutputsWithTweak(
+                spendPublicKey,
+                scanTweak,
+                outputPubKeys
+            );
+    
+            if (matches.size === 0) continue;
+    
+            for (const pubKeyHex of matches.keys()) {
+                const output = outputs.find(
+                    (output) => output.pubKey === pubKeyHex
+                );
+                if (output) {
+                    matchedUTXOs.push(new Coin({
+                        txid: transaction.txid,
+                        vout: output.vout,
+                        value: output.value,
+                        address: encodeAddress(Buffer.from(output.pubKey, 'hex'), this.network.network.bech32),
+                        status: {
+                            isConfirmed: true,
+                        },
+                    }));
+                }
+            }
```

Comment:
in summary, let us iterate through all the silent payment addresses, extract the `scan_key`, `spend_key` and **label_Map**, call the function 
``` typescript
scanOutputs = (
    scanPrivateKey: Buffer,
    spendPublicKey: Buffer,
    sumOfInputPublicKeys: Buffer,
    inputHash: Buffer,
    outputs: Buffer[],
    labels?: LabelMap,
): Map<string, Buffer> 
```
Find the matches, and finally store the coins, with the included `tweaked_data`